parachain/pallets/ethereum-beacon-light-client/src/lib.rs

```diff
@@ -0,0 +1,710 @@
+}
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct LightClientFinalizedHeaderUpdate {
+	pub attested_header: BeaconBlockHeader,
+	pub finalized_header: BeaconBlockHeader,
+	pub finality_branch: ProofBranch,
+	pub sync_aggregate: SyncAggregate,
+	pub fork_version: ForkVersion,
+}
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct ForkData {
+	// 1 or 0 bit, indicates whether a sync committee participated in a vote
+	pub current_version: [u8; 4],
+	pub genesis_validators_root: [u8; 32],
+}
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct SigningData {
+	pub object_root: Root,
+	pub domain: Domain,
+}
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct Genesis {
+	pub validators_root: Root,
+}
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+
+	use super::*;
+
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	use milagro_bls::{AggregatePublicKey, AggregateSignature, AmclError, Signature};
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+	}
+
+	#[pallet::event]
+	pub enum Event<T> {}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		AncientHeader,
+		SkippedSyncCommitteePeriod,
+		SyncCommitteeMissing,
+		Unknown,
+		SyncCommitteeParticipantsNotSupermajority,
+		InvalidSyncCommiteeSignature,
+		InvalidHeaderMerkleProof,
+		InvalidSyncCommitteeMerkleProof,
+		InvalidSignature,
+		InvalidSignaturePoint,
+		InvalidAggregatePublicKeys,
+		InvalidHash,
+		SignatureVerificationFailed,
+		NoBranchExpected,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::storage]
+	pub(super) type FinalizedHeaders<T: Config> =
+		StorageMap<_, Identity, H256, BeaconBlockHeader, OptionQuery>;
+
+	#[pallet::storage]
+	pub(super) type FinalizedHeadersBySlot<T: Config> =
+		StorageMap<_, Identity, u64, H256, OptionQuery>;
+
+	/// Current sync committee corresponding to the active header.
+	/// TODO  prune older sync committees than xxx
+	#[pallet::storage]
+	pub(super) type SyncCommittees<T: Config> =
+		StorageMap<_, Identity, u64, SyncCommittee, ValueQuery>;
+
+	#[pallet::storage]
+	pub(super) type ChainGenesis<T: Config> = StorageValue<_, Genesis, ValueQuery>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig {}
+
+	#[cfg(feature = "std")]
+	impl Default for GenesisConfig {
+		fn default() -> Self {
+			Self {}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(1_000_000)]
+		#[transactional]
+		pub fn initial_sync(
```

Comment:
I get the sense that `initial_sync` will be called once to setup the genesis config of the pallet? Is that right?

If that's the case, then we should be initializing that config using the `GenesisConfig` struct. More details on how to configure genesis config at https://docs.substrate.io/how-to-guides/v3/basics/genesis/

For reference, look at how our 'asset-registry' defines its genesis config:
1. In the pallet: https://github.com/Snowfork/snowbridge/blob/4f407d17a8ebb5dcd1a540c9e8186806ba08a104/parachain/pallets/asset-registry/src/lib.rs#L25
2. actual defaults for genesis config are part of the chainspec: https://github.com/Snowfork/snowbridge/blob/4f407d17a8ebb5dcd1a540c9e8186806ba08a104/parachain/src/chain_spec/snowbase.rs#L114

Reply:
Logged Linear ticket [SNO-263](https://linear.app/snowfork/issue/SNO-263/add-genesisconfig-for-initial-sync) for this.